extensions/git/src/model.ts

```diff
@@ -808,6 +808,16 @@ export class Model implements IRepositoryResolver, IBranchProtectionProviderRegi
 		updateOperationInProgressContext();
 
 		const dispose = () => {
+			// Dispose submodules
+			for (const submodule of repository.submodules) {
+				this.getOpenRepository(path.join(repository.root, submodule.path))?.dispose();
+			}
+
+			// Dispose worktrees
+			for (const worktree of repository.worktrees) {
+				this.getOpenRepository(worktree.path)?.dispose();
+			}
+
```

Comment:
Ah! Now I don't like this 😆 

Let's chat tomorrow and we'll figure out a path forward.